crates/flux-infer/src/fixpoint_encoding.rs

```diff
@@ -1010,11 +1020,32 @@ impl std::str::FromStr for TagIdx {
     }
 }
 
+#[derive(Default)]
+struct ConstEnv<'tcx> {
+    const_map: ConstMap<'tcx>,
+    const_map_rev: HashMap<fixpoint::GlobalVar, ConstKey<'tcx>>,
+}
+
+impl<'tcx> ConstEnv<'tcx> {
+    fn get_or_insert(
+        &mut self,
+        key: ConstKey<'tcx>,
+        make_name: impl FnOnce() -> fixpoint::GlobalVar,
```

Comment:
Oh yikes, I thought we could keep the `global_var_gen` inside this abstraction, but now I realize we also have the `fun_def_map`. 🤔 

I think it makes sense to put the `global_var_gen` and the `fun_def_map` inside `ConstEnv` as well, so there's only one global var dealer.

You can try doing that, but if there's any blocker, I'm fine with the code as is, or you can revert to how it was before.